components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -409,20 +371,47 @@ open class GleanInternalAPI internal constructor () {
         }
     }
 
+    /**
+     * Send a list of pings by name.
+     *
+     * Each ping will be looked up in the known instances of [PingType]. If the
+     * ping isn't known, an error is logged and the ping isn't queued for uploading.
+     *
+     * While the collection of metrics into pings happens synchronously, the
+     * ping queuing and ping uploading happens asyncronously.
+     * There are no guarantees that this will happen immediately.
+     *
+     * If the ping currently contains no content, it will not be sent.
+     *
+     * @param pingNames List of ping names to send.
+     * @return true if any pings had content and were queued for uploading
+     */
+    internal fun sendPingsByName(pingNames: List<String>): Job? {
+        val pings = pingNames.mapNotNull { pingName ->
+            PingType.pingRegistry.get(pingName)?.let {
+                it
+            } ?: run {
+                logger.error("Attempted to send unknown ping '$pingName'")
+                null
```

Comment:
When this happens, we'll have `null` in `pings`. Should we filter for `null`s again?

Reply:
I don't think so, because of the use of [mapNotNull](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/map-not-null.html).

Reply:
Gah, sorry, I misread the docs there.